File: doc/codes/python/examples/api/compute/modulation/fourier-example.py
from autd3.units import Hz
from autd3_modulation import (
    FourierOption,
    SineComponent,
    SineOption,
    fourier,
    modulation_buffer,
)

out = modulation_buffer()

fourier(
    [
        SineComponent(
            freq=100 * Hz,
            option=SineOption(),
        )
    ],
    FourierOption(
        scale_factor=None,
        clamp=False,
        offset=0x00,
    ),
    out,
)

File: doc/codes/rust/examples/api/compute/modulation/fourier-example.rs
use anyhow::Result;

use autd3_rs::units::Hz;
use autd3_rs_modulation::{FourierOption, SineComponent, SineOption, fourier};

fn main() -> Result<()> {
    let mut dst = Vec::new();

    fourier(
        &[SineComponent {
            freq: 100 * Hz,
            option: SineOption::default(),
        }],
        &FourierOption {
            scale_factor: None,
            clamp: false,
            offset: 0x00,
        },
        &mut dst,
    )?;

    Ok(())
}
